contracts/contribute/ContributeManager.sol

```diff
@@ -288,8 +336,19 @@ contract ContributeManager is ERC721TokenReceiver {
             revert WrongServiceSetup(socialId, serviceId, multisig);
         }
 
-        // Transfer the service NFT
-        INFToken(serviceRegistry).safeTransferFrom(msg.sender, address(this), serviceId);
+        // Transfer the service NFT, if owned by contributor
+        address serviceOwner = INFToken(serviceRegistry).ownerOf(serviceId);
```

Comment:
After the service is unstaked, it is either kept by a Contribute Manager, or transferred to a contributor (`msg.sender`)

Reply:
why either or?

Reply:
When switching from one staking contract to another we don't want to let service go, we keep it owned by the CM.

Reply:
The original sin in this entire design was that ContributeManager owns anything. A periphery contract is NOT allowed to have any assets. I will open a PR now that proposes how to fix this going forward. Your current design is still flawed in this way, making it so any further error again leads to very complex recovery logic -if at all possible.